File: JavaScript/src/Internals/Functions/MathBase/Function_FACT.js
import { Function_1 } from '../Function_1.js';
import { Operand } from '../../../Operand.js';

class Function_FACT extends Function_1 {
    get Name() {
        return "Fact";
    }

    constructor(a) {
        super(a);
    }

    Evaluate(engine, tempParameter) {
        let args1 = this.GetNumber_1(engine, tempParameter);
        if (args1.IsError) { return args1; }

        let z = args1.IntValue;
        if (z < 0) {
            return this.FunctionError();
        }
        let d = 1;
        for (let i = 1; i <= z; i++) {
            d *= i;
        }
        return Operand.Create(d);
    }
}

export { Function_FACT };

File: csharp/ToolGood.Algorithm/Internals/Functions/MathBase/Function_FACT.cs
using System;
using System.Text;

namespace ToolGood.Algorithm.Internals.Functions.MathBase
{
	internal sealed class Function_FACT : Function_1
    {
        public Function_FACT(FunctionBase func1) : base(func1)
        {
        }

        public override string Name => "Fact";

        public override Operand Evaluate(AlgorithmEngine engine, Func<AlgorithmEngine, string, Operand> tempParameter)
        {
            var args1 = GetNumber_1(engine, tempParameter);
			if (args1.IsError) { return args1; }

            var z = args1.IntValue;
            if (z < 0) {
                return FunctionError();
            }
            double d = 1;
            for (int i = 1; i <= z; i++) {
                d *= i;
            }
            return Operand.Create(d);
        }

    }

    

}
